src/dreammaker/config.rs

```diff
@@ -0,0 +1,146 @@
+            "info" | "infos" => Ok(WarningLevel::Info),
+            "hint" | "hints" => Ok(WarningLevel::Hint),
+            "false" | "off" => Ok(WarningLevel::Disabled),
+            _other => Ok(WarningLevel::Unset),
+        };
+        warninglevel
+    }
+}
+
+impl From<WarningLevel> for Severity {
+    fn from(warninglevel: WarningLevel) -> Self {
+        match warninglevel {
+            WarningLevel::Error => Severity::Error,
+            WarningLevel::Warning => Severity::Warning,
+            WarningLevel::Info => Severity::Info,
+            WarningLevel::Hint => Severity::Hint,
+            _ => Severity::Error,
+        }
+    }
+}
+
+#[derive(Deserialize, Default)]
+pub struct Config {
+    #[serde(default)]
+    pub warnings: Warnings,
+}
+
+impl Config {
+    pub fn new() -> Config {
+        Config {
+            warnings: Warnings::default()
+        }
+    }
+}
+
+#[derive(Deserialize, Default, Debug, Clone)]
+pub struct Warnings {
+    #[serde(default)]
+    duplicate_includes: WarningLevel,
+
+    #[serde(default)]
+    should_call_parent: WarningLevel,
+
+    #[serde(default)]
+    should_not_override: WarningLevel,
+
+    #[serde(default)]
+    field_access_static_type: WarningLevel,
+
+    #[serde(default)]
+    proc_call_static_type: WarningLevel,
+
+    #[serde(default)]
+    integer_precision_loss: WarningLevel,
+
+    #[serde(default)]
+    colon_path_warning: WarningLevel,
+
+    #[serde(default)]
+    var_in_proc_paramater: WarningLevel,
+
+    #[serde(default)]
+    static_in_proc_paramater: WarningLevel,
+
+    #[serde(default)]
+    macro_redefined: WarningLevel,
+
+    #[serde(default)]
+    undefine_undefined_macro: WarningLevel,
+
+    #[serde(default="Severity::default_all")]
+    error_level: Severity,
+}
+
+impl Warnings {
+    pub fn warning_level_for(&self, errortype: &'static str) -> WarningLevel {
+        match errortype {
+            "duplicate_include" => { return self.duplicate_includes },
+            "must_call_parent" => { return self.should_call_parent },
+            "must_not_override" => { return self.should_not_override },
+            "field_access_static_type" => { return self.field_access_static_type },
+            "proc_call_static_type" => { return self.proc_call_static_type },
+            "integer_precision_loss" => { return self.integer_precision_loss },
+            "colon_path_warning" => { return self.colon_path_warning },
+            "var_in_proc_paramater" => { return self.var_in_proc_paramater },
+            "static_in_proc_parameter" => { return self.var_in_proc_paramater },
+            "macro_redefined" => { return self.macro_redefined },
+            "undefine_undefined_macro" => { return self.undefine_undefined_macro }
+            _ => { return WarningLevel::Unset }
+        };
+    }
+
+    pub fn severe_enough(&self, severity: Severity) -> bool {
+        return severity <= self.error_level
+    }
+}
+
+pub fn read_config_toml(path: String) -> Config {

```

Comment:
Error handling is kind of a mess in this function, perhaps return a Result and let the caller deal with it